intents.go

```diff
@@ -44,6 +46,33 @@ type Stake struct {
 	ChainId string    `json:"chainId"`
 }
 
+type Supply struct {
+	Type AssetType `json:"type,omitempty" binding:"required"`
+
+	// Asset is the contract address for the token to Supply or add into the
+	// Protocol
+	Asset string `json:"asset,omitempty" binding:"required"`
+
+	// can be empty? and solver chooses a default protocol to supply to?
+	// this would be the Contract address for the Protocol
+	// Aave3 as an example would be 0x87870Bca3F3fD6335C3F4ce8392D69350B4fA4E2
+	Protocol string `json:"protocol,omitempty"`
```

Comment:
Now i understand that it is a contract address? Why not a Geth address type? Also the name should indicate that.

Reply:
I was following existing types we use for addresses and others. I would change that.

Reply:
you're right about the address data type. We're using a string data type in the model.